File: Tema5/dockerlexer/java/Main.java
package dockerlexer;

import java.nio.file.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        if (args.length < 1) {
            System.err.println("Uso: java dockerlexer.Main <archivo.dockerfile>");
            System.exit(1);
        }

        String content = Files.readString(Paths.get(args[0]));
        List<Parser.Node> ast = Parser.parseDockerfile(content);

        StringBuilder sb = new StringBuilder();
        sb.append("[\n");
        for (int i = 0; i < ast.size(); i++) {
            sb.append("  ").append(ast.get(i).toJson());
            if (i < ast.size() - 1) sb.append(",");
            sb.append("\n");
        }
        sb.append("]");
        System.out.println(sb);
    }
}

File: Tema5/dockerlexer/rust/src/main.rs
mod lexer;
mod parser;

use std::env;
use std::fs;
use std::process;

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() < 2 {
        eprintln!("Uso: {} <archivo.dockerfile>", args[0]);
        process::exit(1);
    }

    let content = fs::read_to_string(&args[1]).unwrap_or_else(|e| {
        eprintln!("Error leyendo archivo: {}", e);
        process::exit(1);
    });

    match parser::parse_dockerfile(&content) {
        Ok(ast) => {
            let items: Vec<String> = ast.iter().map(|n| format!("  {}", n.to_json())).collect();
            println!("[\n{}\n]", items.join(",\n"));
        }
        Err(e) => {
            eprintln!("Error de parseo: {}", e.0);
            process::exit(1);
        }
    }
}
